### PR TITLE
[Hub Generated] Review request for Microsoft.Insights to add version stable/2020-10-01

### PR DESCRIPTION
**Pull request by InbalZim**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/InbalZim/azure-rest-api-specs/dev-monitor-Microsoft.Insights-2020-10-01...master/).

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “**WaitForARMFeedback**” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ravbhatnagar** [comment]
Signing off from ARM side

**Comment by InbalZim** [comment]
> Signing off from ARM side

**Comment by InbalZim** [comment]
@kairu-ms Can you please review and approve the PR now that ARM had sign off on it?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json [hunk 1]
```diff
@@ -0,0 +1,540 @@
...
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/activityLogAlerts/{alertRuleName}": {
+      "put": {
+        "description": "Create a new Activity Log Alert rule or update an existing one.",
+        "operationId": "ActivityLogAlertRule_CreateOrUpdate",
```
Comment (kairu-ms) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json:43]: Why you change operationId prefix? It's a breaking change for Python SDK. @msyyc FYI.

Comment (InbalZim) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json]: Reverted the operationId change.

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json [hunk 2]
```diff
@@ -0,0 +1,540 @@
...
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/activityLogAlerts/{alertRuleName}": {
```
Comment (kairu-ms) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json:40]: Can you change alertRuleName back to activityLogAlertName ?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json [hunk 3]
```diff
@@ -0,0 +1,540 @@
...
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/AlertRuleNameParameter"
```
Comment (kairu-ms) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json:63]: Can you rename back?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json [hunk 4]
```diff
@@ -0,0 +1,540 @@
...
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "activityLogAlertRule",
+            "description": "The Activity Log Alert rule to create or use for the update.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/AlertRuleResource"
```
Comment (kairu-ms) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json:74]: Can you rename back?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json [hunk 5]
```diff
@@ -0,0 +1,540 @@
...
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/AlertRuleNameParameter"
```
Comment (kairu-ms) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json:115]: Can you rename back?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json [hunk 6]
```diff
@@ -0,0 +1,540 @@
...
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request succeeded.",
+            "schema": {
+              "$ref": "#/definitions/AlertRuleResource"
```
Comment (kairu-ms) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json:125]: Can you rename back?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json [hunk 7]
```diff
@@ -0,0 +1,540 @@
...
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/AlertRuleNameParameter"
```
Comment (kairu-ms) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json:152]: Can you rename back?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json [hunk 8]
```diff
@@ -0,0 +1,540 @@
...
```
Comment (kairu-ms) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json:189]: Can you rename back?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json (hunk 2 above)
Comment (kairu-ms) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json:40]: 'microsoft.insights' is changed to 'Microsoft.Insights'. It's a breaking change. @jianyexi FYI.

**Comment by kairu-ms** [comment]
@InbalZim You should rename back parameter and definition names. If you want to insist on changing names, you need to follow up breaking change policy.

**Comment by kairu-ms** [comment]
@jianyexi Could your help with lint issues?

**Review by kairu-ms — APPROVED** [review]

**Comment by InbalZim** [comment]
Waiting for RP manifest to be updated, before merging the swagger changes

**Comment by jianyexi** [comment]
@InbalZim For the lintDiff error: RequiredSystemDataInNewApiVersions , it's newly asked by ARM , you'd better support it.
see https://armwiki.azurewebsites.net/api_contracts/ResourceSystemData.html

**Comment by napolish** [comment]
@kairu-ms Do we need to fix the lintDiff  and SDK errors? Or when we are ready it is possible to merge the changes?

**Comment by kairu-ms** [comment]
> @kairu-ms Do we need to fix the lintDiff and SDK errors? Or when we are ready it is possible to merge the changes?

You should fix lintDiff error. It blocks mergence.

**Comment by kairu-ms** [comment]
@InbalZim Please fix Conflicting files `specification/monitor/resource-manager/readme.md`

**Pull request merged**